### PR TITLE
[Net][GUI][RPC] Add enable/disable network activity feature

### PR DESCRIPTION
**Pull request by furszy**
Allowing the user/test to stop and restart the network activity.

Pretty useful for some manual isolation test cases.

When the network activity is disabled the client will close all connections, stop accepting inbound connections, and stop opening new outbound connections, until the network activity is reenabled.

Quick demo video:

https://user-images.githubusercontent.com/5377650/137732464-02624285-2e62-483d-bf79-6df9cb0162f1.mp4

TODO:
Add release-notes for the new RPC command and the GUI control.

### CODE REVIEW

**Comment by furszy** [comment]
rebased on master, conflicts solved.

**Comment by furszy** [comment]
rebased, conflicts solved.

**Review by Fuzzbawls — APPROVED** [review]
ACK 1f21536fd744a2153757a396ea1e14b765bfe80c

File: src/qt/rpcconsole.cpp [hunk 1]
```diff
@@ -836,3 +859,7 @@ void RPCConsole::showOrHideBanTableIfRequired()
     ui->banHeading->setVisible(visible);
 }
 
+void RPCConsole::on_toggleNetworkActiveButton_clicked()
```
Comment (random-zebra) [inline src/qt/rpcconsole.cpp:862]: this is not connected to any button, isn't it?

**Review by random-zebra — COMMENTED** [review]
Looking good. One minor thing: the checked state of the buttons `pushNetEnable`/`pushNetDisable` should change if the activity has been changed via RPC.

File: src/qt/rpcconsole.cpp (hunk 1 above)
Comment (furszy) [reply src/qt/rpcconsole.cpp:862]: yeah, not connected. Good, cleaning

File: src/qt/rpcconsole.cpp [hunk 2]
```diff
@@ -436,16 +439,36 @@ void RPCConsole::message(int category, const QString& message, bool html)
...
+
 void RPCConsole::setNumConnections(int count)
 {
     if (!clientModel)
         return;
 
-    QString connections = QString::number(count) + " (";
-    connections += tr("In:") + " " + QString::number(clientModel->getNumConnections(CONNECTIONS_IN)) + " / ";
-    connections += tr("Out:") + " " + QString::number(clientModel->getNumConnections(CONNECTIONS_OUT)) + ")";
+    updateNetworkState();
```
[the remark this answers is not in the document]
Comment (furszy) [reply src/qt/rpcconsole.cpp:466]: sure

**Comment by furszy** [comment]
done, updated per feedback.

**Review by random-zebra — APPROVED** [review]
ACK 102dcafa654acfc884411905ba7b134d121685c0

**Review by Fuzzbawls — APPROVED** [review]
ACK 102dcafa654acfc884411905ba7b134d121685c0

**Pull request merged**